vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -318,25 +385,49 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 		for _, mount := range containerJSONInfo.Mounts {
 			// check if the mount location belongs to vsphere plugin
 			if matchNameforVMDK(mount.Source) {
-				r.Incr(mount.Name)
-				log.Debugf("  name=%v (driver=%s source=%s) (%v)",
+				// populate temp map
+				rc := r.auxRefMap[mount.Name]
+				if rc == nil {
+					rc = newRefCount()
+					r.auxRefMap[mount.Name] = rc
+				}
+				rc.count++
+
+				log.Infof("  name=%v (driver=%s source=%s) (%v)",
 					mount.Name, mount.Driver, mount.Source, mount)
 			}
 		}
 	}
 
-	// Check that refcounts and actual mount info from Linux match
-	// If they don't, unmount unneeded stuff, or yell if something is
-	// not mounted but should be (it's error. we should not get there)
+	// check if the background refcount was dirtied
+	// if not, copy auxRefMap into refMap and unmount unwanted volumes
+	r.curr_state.mtx.Lock()
+	if r.curr_state.dirty == false {
+		for k, v := range r.auxRefMap {
+			r.refMap[k] = v
+		}
 
-	r.getMountInfo()
-	r.syncMountsWithRefCounters(d)
+		// Check that refcounts and actual mount info from Linux match
+		// If they don't, unmount unneeded stuff, or yell if something is
+		// not mounted but should be (it's error. we should not get there)
+		r.getMountInfo()
```

Comment:
Yes. In this section where state si clean, no mount/unmount can enter since they will wait to make state dirty. So we can proceed to manipulate the actual refmap